kernel/enter.c

```diff
@@ -95,6 +95,83 @@ static int _process_mount_configuration(myst_mounts_config_t* mounts)
     return ret;
 }
 
+static int _copy_host_etc_resolv()
+{
+    int ret = 0, size, fd = -1;
+    const char* resolv_file = "/etc/resolv.conf";
+    char* buf = NULL;
+    struct stat statbuf;
+
+    if ((size = myst_tcall_get_file_size(resolv_file)) < 0)
+    {
+        myst_eprintf("kernel: failed to get file size %s\n", resolv_file);
+        ERAISE(-EINVAL);
+    }
+
+    if (!(buf = malloc(size + 1)))
+        ERAISE(-ENOMEM);
+
+    if ((myst_tcall_read_file(resolv_file, buf, size)) < 0)
+    {
+        myst_eprintf("kernel: failed to read file %s\n", resolv_file);
+        ERAISE(-EINVAL);
+    }
+    buf[size] = 0;
```

Comment:
I had this additional null termination for /proc files copied from the host, as linux doesn't terminate them properly. Not sure if thats the case with /etc/resolv.conf.

Reply:
Good point! I thought we need the trailing null to print buf as a string. Changed since we don't need it.